Modules/_asynciomodule.c

```diff
@@ -1587,6 +1692,19 @@ FutureIter_am_send(futureiterobject *it,
     return PYGEN_ERROR;
 }
 
+static PySendResult
+FutureIter_am_send(futureiterobject *it,
+                   PyObject *Py_UNUSED(arg),
+                   PyObject **result)
+{
+    PySendResult res;
+    Py_BEGIN_CRITICAL_SECTION2(it, it->future);
```

Comment:
`it->future` may be NULL, which would crash I think.

Also, the locking here doesn't seem right, but I'm not really sure what the correct behavior should be.

Reply:
> it->future may be NULL, which would crash I think.

fixed

Reply:
> Also, the locking here doesn't seem right, but I'm not really sure what the correct behavior should be.

The locking is required because there can be multiple awaiters for one future and locking is required for getting the result from the future.